tests/Features/Contexts/FeatureContext.php

```diff
@@ -0,0 +1,216 @@
+    {
+        $this->visit('signup');
+    }
+
+    /**
+     * @When I agree to the code of conduct
+     */
+    public function iAgreeToTheCodeOfConduct()
+    {
+        $this->checkOption('coc');
+    }
+
+    /**
+     * @Then I am informed about a failed signup
+     */
+    public function iAmInformedAboutAFailedSignup()
+    {
+        $this->assertPageContainsText('there was an error');
+    }
+
+    /**
+     * @Given I have signed up
+     */
+    public function iHaveSignedUp()
+    {
+        /** @var AccountManagement $accountManagement */
+        $accountManagement = $this->getContainer()->get(AccountManagement::class);
+
+        $accountManagement->create($this->accountInfo['email'], $this->accountInfo['password']);
+    }
+
+    /**
+     * @Given my account is active
+     */
+    public function myAccountIsActive()
+    {
+        /** @var AccountManagement $accountManagement */
+        $accountManagement = $this->getContainer()->get(AccountManagement::class);
+        $accountManagement->activate($this->accountInfo['email']);
+    }
+
+    /**
+     * @Given I am on the login page
+     */
+    public function iAmOnTheLoginPage()
+    {
+        $this->visit('login');
+    }
+
+    /**
+     * @When I fill in my account information
+     */
+    public function iFillInMyAccountInformation()
+    {
+        $this->fillFields(new TableNode([
+            ['email', $this->accountInfo['email']],
+            ['password', $this->accountInfo['password']],
+        ]));
+    }
+
+    /**
+     * @When I login
+     */
+    public function iLogin()
+    {
+        $this->pressButton('Login');
+    }
+
+    /**
+     * @Then I should be guided to start my profile
+     */
+    public function iShouldBeGuidedToStartMyProfile()
+    {
+        $this->assertPageContainsText('Fill out your profile');
+    }
+
+    /**
+     * @Given I have previously completed my profile
+     */
+    public function iHavePreviouslyCompletedMyProfile()
+    {
+        $user = User::where('email', $this->accountInfo['email'])->first();
+        $user->update([
+            'first_name'       => 'Some',
+            'last_name'        => 'Speaker',
+            'has_made_profile' => 1,
+        ]);
+    }
+
+    /**
+     * @Then I should be guided to submit my first talk
+     */
+    public function iShouldBeGuidedToSubmitMyFirstTalk()
+    {
+        $this->assertPageContainsText('Get Started');
+        $this->assertPageContainsText('Submit a Talk');
+    }
+
+    /**
+     * @Given I have logged in
+     */
+    public function iHaveLoggedIn()
+    {
+        $this->iAmOnTheLoginPage();
+        $this->iFillInMyAccountInformation();
+        $this->iLogin();
+    }
+
+    /**
+     * @Given I am on the dashboard
+     */
+    public function iAmOnTheDashboard()
+    {
+        $this->visit('dashboard');
+    }
+
+    /**
+     * @When I submit a talk about :title
+     * @Given I have submitted a talk about :title
```

Comment:
Maybe this should be split up in two functions

`@When` for doing the action
`@Given` for simply updating the database